### PR TITLE
added login logout api and fixes issues

### PR DESCRIPTION
**Pull request by pyarati**

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -2,16 +2,22 @@
...
 
 app = Flask(__name__)
 app.config['SECRET_KEY'] = '\x1f\x19\xc7\x95\xb6\xac\xd9\x1c\xbd\xd8%V\xd8\x1b@\xdf!\x13A\x9eW8\xa7\xc0'
 app.config['SQLALCHEMY_DATABASE_URI'] = 'mysql+pymysql://root:mySQL@123@localhost/bank_system'
 app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
 app.config["DEBUG"] = True
-
+app.config["JWT_SECRET_KEY"] = '1234567890abcdefghijklmnopqrstuvwxyz'
+app.config['JWT_COOKIE_CSRF_PROTECT'] = True
+app.config['JWT_CSRF_CHECK_FORM'] = True
```
Comment (shreyas-damle) [inline app/__init__.py]: What are these?

File: app/models/tokenblocklist.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from app import db
+
+
+class TokenBlockList(db.Model):
+    __tablename__ = "TokenBlockList"
+    id = db.Column(db.Integer, primary_key=True)
+    jti = db.Column(db.String(36), nullable=False)
```
Comment (shreyas-damle) [inline app/models/tokenblocklist.py:7]: What is `jti`?

File: app/schemas/account.py [hunk 3]
```diff
@@ -2,18 +2,18 @@
 from app import ma
 from marshmallow.validate import Length, Regexp
 
-string_pattern = "^[a-zA-Z]*$"
+string_pattern = "^[a-zA-Z ]*$"
```
Comment (shreyas-damle) [inline app/schemas/account.py:5]: Why additional space here?

File: app/views/login_logout.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                 404:
+                     description: User email_id and password not exist
+                 201:
+                     description: Successfully logged in
+                 400:
+                     description: Bad request
+        """
+        try:
+            email_id = request.json.get("email_id", None)
+            password = request.json.get("password", None)
```
Comment (shreyas-damle) [inline app/views/login_logout.py]: Why are you setting default value to `None`?
Do not set it to `None` and handle `KeyError`

File: app/views/user.py [hunk 5]
```diff
@@ -7,6 +7,31 @@
...
+from functools import wraps
+from flask_jwt_extended.exceptions import NoAuthorizationError
+
+
+def custom_validator(view_function):
+    @wraps(view_function)
+    def wrapper(*args, **kwargs):
+        jwt_data = decode_token(encoded_token='access_token')
+
+        # Do your custom validation here.
```
Comment (shreyas-damle) [inline app/views/user.py]: Remove the comment

**Review by shreyas-damle — CHANGES_REQUESTED** [review]

File: app/__init__.py (hunk 1 above)
Comment (pyarati) [reply app/__init__.py]: no use of this, so removed it

Comment (pyarati) [reply app/__init__.py]: Made the changes as suggested.

File: app/schemas/account.py (hunk 3 above)
Comment (pyarati) [reply app/schemas/account.py:5]: It accept string with space between 2 words

Comment (pyarati) [reply app/schemas/account.py:5]: Made the changes as suggested.

File: app/views/user.py (hunk 5 above)
Comment (pyarati) [reply app/views/user.py]: Ok sir

Comment (pyarati) [reply app/views/user.py]: Made the changes as suggested.

File: app/models/tokenblocklist.py (hunk 2 above)
Comment (pyarati) [reply app/models/tokenblocklist.py:7]: jti is encoded token like email_id and password and we storing in table TokenBlockList, so this token can not be reuse again

File: app/views/login_logout.py (hunk 4 above)
Comment (pyarati) [reply app/views/login_logout.py]: Made the changes as suggested.

File: app/models/tokenblocklist.py (hunk 2 above)
Comment (pyarati) [reply app/models/tokenblocklist.py:7]: Made the changes as suggested.

**Pull request merged**